### PR TITLE
Azure AI Studio Custom Skill for Text Summarization

### PR DESCRIPTION
**Pull request by amitkalay**
## Purpose

* This PR adds our very first custom skill using Azure AI Studio! The code added here illustrates a call into our AOAI deployment to do text summarization. Detailed documentation on how to run the code is provided in the README.

## Does this introduce a breaking change?

```
[ ] Yes
[x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Information provided in the README

## Test Plan

I sent a request to the /api/summarize endpoint and got the following output:
```
HTTP/1.1 200 OK
Date: Fri, 04 Oct 2024 20:29:30 GMT
Content-Type: application/json
Transfer-Encoding: chunked
Connection: close
Cache-Control: no-cache,no-store
Expires: Thu, 01 Jan 1970 00:00:00 GMT
Pragma: no-cache
Set-Cookie: .Tunnels.Relay.WebForwarding.Cookies=; expires=Thu, 01 Jan 1970 00:00:00 GMT; path=/; secure; samesite=none,.Tunnels.Relay.WebForwarding.Cookies=CfDJ8E0FHi1JCVNKrny-ARCYWxM-D2lCVBfcNJJ-AoLveoZR0EYildsrrCG7Cdy531e5soHN8de1cxCc4of1aVs8PbsPdvs0NVR7HXoHsEd3ZG4Pb2NdDa82NFh13mgjij4ZD6vgg-3LgFA9mt8vyfecccrcTfur9yABsFpC5hHvJGwWE9XpytKXe__-WkLI2B8ykPv3nn5k8v1tAqP_pQ81sgG_oMsOfnk-KNOYwmE1gy0yZ06qgav4B5GBETwBCrujlw7M5hddCgTnZIDUC9rXVto1L-pSONpYXJlahd5slX5FKHI7MHrHP279IVaoU1U2EP7NXQlPg_Ef3s0C4mN274AMOKoqObaXi_Bh80ZaDFAdmlGmDbq9iz6Qc29r45F_8VN8R2v7mJcdLAPUdwBNCFIomF4OvBJ1uURYmgkOgdlkBjX5fJGC8PddoXEzFscBukwuDv4OjtLaw6N96bHoREZqWpeUfkgvcCCvg9rmPUAa0ga6jrFaVYYV6OXDGTyEzudDqJptrfmJdW6oocUaBAic5OmuYGIrF_vu4OyqlAFWp0CtO-zttIBVN-vEgEGhxlKrk338T4YdjfMbnSx_Q6jgEs5hGNCE4GSSYUJG4ChDYs_RZUB66NZcmS_NCmNr81sMqvl2ZB1thL6U2m7AjxDWRmhITfOtL9yYJPQ-xP4ZzJiYC_EA7Sf3YAc3vXWGLtqBRMSx8iTHp6rzQYTDy7qllDMreDqB87w82x2t5_2s22fjAKH_SgD6_lqpRj2FWgTqBUl8nHRQLeJ0Nka_tOhYID-favouLHUdudgI7svjlASu9pNlFBNHGIIeqQ-irKalqahAxQ7cLtS0-CmC02kS_GGd_ZGSCofiV1GBf7SYvONW6emsAbWZxoIQi__9Aav9s75gQ_s6bV_-KO0iodSoWjzLEJw3XNvvuR1QYPTCLUeW17_XucQQ-j7Hk5E4qfKN5V7EEV0U-KW8P_10nf4JozTF_GvT63DyNHCAO7n9e6qObnC2ZSDtD7YR-7BCfg; path=/; secure; samesite=none
X-Content-Type-Options: nosniff
RateLimit-Limit: HttpRequestRatePerPort:1500/m
RateLimit-Remaining: HttpRequestRatePerPort:1499
RateLimit-Reset: HttpRequestRatePerPort:45s
X-Report-Abuse: https://msrc.microsoft.com/report/abuse
x-ms-ratelimit-limit: 1500
x-ms-ratelimit-remaining: 1499
x-ms-ratelimit-used: 1
x-ms-ratelimit-reset: 
X-Robots-Tag: noindex, nofollow
Referrer-Policy: same-origin
VsSaaS-Request-Id: d80210cc-3748-436c-b191-59cbfd5c5201
Strict-Transport-Security: max-age=31536000; includeSubDomains
X-Served-By: tunnels-prod-rel-usw3-v3-cluster

{
  "values": [
    {
      "warnings": null,
      "errors": [],
      "recordId": "0",
      "data": {
        "generative-summary": "OpenAI has secured a $6.6 billion investment at a $157 billion valuation to advance its mission of developing artificial general intelligence (AGI). The funding round, led by Thrive Capital with a $1 billion commitment, includes a unique provision allowing Thrive to invest an additional $1 billion next year at the same valuation if OpenAI meets a revenue target. The funds hinge on OpenAI restructuring into a for-profit entity within two years, otherwise, investors can reclaim their money. OpenAI is also contemplating becoming a public benefit corporation. Additionally, OpenAI has requested investors avoid supporting rival AI firms like Anthropic and Elon Musk's xAI. The significant investment will support the high costs of training advanced AI models, with future models potentially costing up to $100 billion to develop."
      }
    }
  ]
}
```

### CODE REVIEW

File: AIStudioTextSummarization/README.md [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
 
-Description:
+Quickstart Guide:
 
-- This custom skill enables you to summarize text using Azure AI Studio. You will pass in the text you want to summarize along with a prompt for [Azure AI Studio](https://learn.microsoft.com/en-us/azure/ai-studio/what-is-ai-studio). We will update this README as the features get completed!
+- This folder illustrates how to leverage Azure AI Studio to summarize a large piece of text using [custom web api skills](https://learn.microsoft.com/en-us/azure/search/cognitive-search-custom-skill-web-api).
+
+- As a prerequisite to running this custom skill, you must first deploy an LLM to Azure. See [this guide](https://learn.microsoft.com/en-us/azure/ai-studio/how-to/deploy-models-openai) for reference.
+
+- in the *local.settings.json* file, set your "AZURE_INFERENCE_CREDENTIAL" value to the API key of your deployed LLM.
```
Comment (bleroy) [inline AIStudioTextSummarization/README.md:11]: Can we make it work with MI?

File: AIStudioTextSummarization/README.md [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
+
+- Once you are in this folder, you need to install the Azure functions extension and update the core tools. After that, you should run func start as descibed in this [python quickstart for Azure functions](https://learn.microsoft.com/en-us/azure/azure-functions/create-first-function-cli-python?tabs=windows%2Cbash%2Cazure-cli%2Cbrowser).
+
+- A sample payload is provided in the *api-test.http* file. Replace the localhostBaseUrl variable with the base url of your container/local http environment. Once you install the REST Client extension, you can hit the Send Request button from that file.
```
Comment (bleroy) [inline AIStudioTextSummarization/README.md:15]: Extension to VS Code I assume. If so, please say so explicitly.

**Review by bleroy — APPROVED** [review]

File: AIStudioTextSummarization/README.md (hunk 1 above)
Comment (amitkalay) [reply AIStudioTextSummarization/README.md:11]: What is MI?

File: AIStudioTextSummarization/README.md (hunk 2 above)
Comment (amitkalay) [reply AIStudioTextSummarization/README.md:15]: will do!

File: AIStudioTextSummarization/README.md (hunk 1 above)
Comment (bleroy) [reply AIStudioTextSummarization/README.md:11]: Managed identity

**Pull request merged**